ReactAndroid/src/main/java/com/facebook/react/modules/core/JavaTimerManager.java

```diff
@@ -373,4 +373,23 @@ public void run() {
           }
         });
   }
+
+  /**
+   * Returns a bool representing whether there are any active timers that will be fired within a
+   * certain period of time. Disregards repeating timers (setInterval). Used for testing to
+   * determine if RN is idle.
+   *
+   * @param rangeMs The time range, in ms, to check
+   * @return True if there are pending timers within the given range; false otherwise
+   */
+  /* package */ boolean hasActiveTimersInRange(long rangeMs) {
+    synchronized (mTimerGuard) {
+      for (Timer timer : mTimers) {
```

Comment:
Hey @ejanzer! I hope I'm not too late - seeing this has already been merged, I'd like to request some changes nevertheless.

Here, in order to evaluate whether _active_ timers exist, the timers collection gets iterated over. If you take a closer look into detox' implementation, we _do_ eventually run the same logic, but in addition we [first attempt to inspect the first (i.e. _least_) timer in the collection](https://github.com/wix/Detox/blob/4f81a77baee4e4542a693922bcbde621d9d8c1a5/detox/android/detox/src/main/java/com/wix/detox/reactnative/idlingresources/TimersIdlingResource.kt#L87). That is done for a reason:

As the collection is in fact a `PriorityQueue`, these facts hold true:
1. There's no indexed access to items in it - only access to the _least_ item using `peek()` and `poll()`.
2. (To quote from PriorityQueue.iterator())  _The iterator does not return the elements in any particular order_.

These 2 combined evidently mean that most of the time, the least item is in fact the one we're looking for, as its interval is the lowest. We therefore resolve to this order-less iteration only "if we have to", namely, if the least item happens to be a repeating one.
> Actually, I see now that an additional optimization can be applied: if the least item is a nonrepeating one that has a long interval, the iteration can be avoided as well, as all other timers' intervals are longer by-design.

So, given this tedious explanation..... do you think you could apply these existing optimizations to your implementation? 😄

Reply:
Ah I see, I should have looked at that a little more carefully! So to reiterate, we're going to skip iterating over the entire timers queue if the least timer (timers.peek()) is nonrepeating and its interval is greater than the specified threshold.

That's the only change to make, right? Or were there any other optimizations that I missed?

Reply:
Yes! expect for:
> interval is greater than the specified threshold
interval should be lower (i.e. within the time-window's bounds)

In any case, I'm gonna go ahead and merge + release the Detox changes so you could start working with an actual Detox that does what you need!

Reply:
Yes! except for:
> interval is greater than the specified threshold

interval should be lower (i.e. within the time-window's bounds)

In any case, I'm gonna go ahead and merge + release the Detox changes so you could start working with an actual Detox that does what you need!

Reply:
Yes! except for:

> interval is greater than the specified threshold

interval should be lower (i.e. within the time-window's bounds)

In any case, I'm gonna go ahead and merge + release the Detox changes so you could start working with an actual Detox that does what you need!

Reply:
Thanks @d4vidi! I'll send a PR today with this change.

Reply:
Adding in #27841